### ISSUE TITLE
Train arch_parameters with validation data?

### ISSUE THREAD

**Issue by Harry-Up** [quark0/darts#87]
Thank you for sharing such beautiful implement.

I'm sorry, but I have a question about why arch_parameters are trained with validation data.
In ./cnn/architect, the _backward_step function computes loss for arch_parameter updates with validation data (i.e. input_valid, target_valid). Should not we train arch_parameters with train data and search the proper architecture on validation data?

**Comment by Harry-Up** [comment]
ahh... I just make a misunderstanding. The alpha_normal / alpha_reduce should be the actions to search.

**Comment by marsggbo** [comment]
> Thank you for sharing such beautiful implement.
> 
> I'm sorry, but I have a question about why arch_parameters are trained with validation data.
> In ./cnn/architect, the _backward_step function computes loss for arch_parameter updates with validation data (i.e. input_valid, target_valid). Should not we train arch_parameters with train data and search the proper architecture on validation data?

Good question! I'm also confused with this question. Could you please explain more? I think the validation set should be not be used to update the architecture parameters.

**Comment by Harry-Up** [comment]
Well, the same misunderstanding. The code shows that the architecture weights are updated with training data, and the hyperparameters (i.e., alpha_normal / alpha_reduce) are searched on validation data.

**Comment by xperzy** [comment]
> Well, the same misunderstanding. The code shows that the architecture weights are updated with training data, and the hyperparameters (i.e., alpha_normal / alpha_reduce) are searched on validation data.

See the **val_queue** in the code, which is using part of the training data, not the real validation data.

**Comment by xperzy** [comment]
See the code where defines the dataset, 
```python
dset.CIFAR10(..., train=True ... )
```
This is using the training data, and the `valid_queue` is only part of the training data.
